controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_webhook.go

```diff
@@ -348,7 +348,7 @@ func validateRolloutStrategy(rolloutStrategy *RolloutStrategy, replicas *int32,
 		)
 	}
 
-	if *rolloutStrategy.RollingUpdate.MaxSurge != ios1 && *rolloutStrategy.RollingUpdate.MaxSurge != ios0 {
+	if rolloutStrategy.RollingUpdate.MaxSurge.IntValue() != ios1.IntValue() && rolloutStrategy.RollingUpdate.MaxSurge.IntValue() != ios0.IntValue() {
```

Comment:
I came across this when writing an external patch with Runtime SDK. The variables are stored as strings and using intorstr.FromString() failed this check. Using IntValue means that this only compares the valid int values of the intOrStr

Reply:
I think you're fix is an improvement, but I wonder why we are using IntOrString. Part of the godoc of the field:
> Value can be an absolute number 1 or 0.

I would have assumed based on the type that you could also use something like 50%

Reply:
Looks like we just use it as int value: https://github.com/kubernetes-sigs/cluster-api/blob/3f8d00953b630b329c8d5415ba1646ded73bf068/controlplane/kubeadm/internal/controllers/upgrade.go#L118

@vincepri @fabriziopandini @enxebre Do you know why this field is not simply an int? Are we missing something?

Reply:
Looks like the field technically supports using values like `50%` and to get the scaled value there is a helper function in apimachinery called `GetScaledValueFromIntOrPercent`. 
Also, we are already use this with MaxSurge in MachineDeployments: https://github.com/killianmuldoon/cluster-api/blob/cc27d257e4aa50b05473b1be8c3380e9f2ce108c/internal/controllers/machinedeployment/mdutil/util.go#L515

Reply:
Yup. But as far as I could see we only ever use the int in KCP (maybe I missed a usage though)

Reply:
hmm, I'm not sure, I guess the type might have been initially chosen to enable a possible future smooth transition towards loosing the 1 / 0 requirement and add the ability to input other values for IntOrString consistently with other valid use cases e.g machineDeployment.
I'm not sure we'll ever want to support something else for control pane though.